### PR TITLE
Adjust interval

### PR DESCRIPTION
**Pull request by andy128k**
intention is to make __archive_fetch more predictable. Caller now adjusts interval, so it can predict result of invocation.

### CODE REVIEW

File: whisper/src/lib.rs [hunk 1]
```diff
@@ -586,50 +586,67 @@ fn file_fetch<F: Read + Seek>(fh: &mut F, header: &WhisperMetadata, archive: &Ar
...
         .map_err(|s| io::Error::new(io::ErrorKind::Other, s))?;
 
-    __archive_fetch(fh, archive, interval.from(), interval.until()).map(Some)
+    let adjusted_interval = adjust_interval(&interval, archive.seconds_per_point)
+        .map_err(|s| io::Error::new(io::ErrorKind::Other, s))?;
+
+    __archive_fetch(fh, archive, adjusted_interval).map(Some)
+}
+
+fn adjust_instant(instant: u32, step: u32) -> u32 {
```
Comment (AnderEnder) [inline whisper/src/lib.rs]: It is not clear what this function should do.

Comment (AnderEnder) [reply whisper/src/lib.rs]: Looks like it is a discrete adjust to the left of the step

File: whisper/src/lib.rs [hunk 2]
```diff
@@ -586,50 +586,67 @@ fn file_fetch<F: Read + Seek>(fh: &mut F, header: &WhisperMetadata, archive: &Ar
...
+    instant - (instant % step)
+}
+
+fn adjust_interval(interval: &Interval, step: u32) -> Result<Interval, String> {
+    let from_interval = adjust_instant(interval.from(), step) + step;
```
Comment (AnderEnder) [inline whisper/src/lib.rs]: Should it be always + step?
IMHO it should be:
* interval.from() if (instant % step) == 0
* adjust_instant(interval.from(), step) +step if  (instant % step) > 0

So in general I would move this to adjust_instant as:
```rust
fn adjust_instant(instant: u32, step: u32) -> u32 {
    let diff = (instant % step);
    if diff > 0 {
        instant - diff + step
    }
    else {
        instant
   }
}
```

or even use arithmetics like:
``` rust
fn adjust_instant(instant: u32, step: u32) -> u32 {
    let diff = (instant % step);
    instant - diff + (diff/step).ceil() * step
}
```

File: whisper/src/lib.rs (hunk 1 above)
Comment (andy128k) [reply whisper/src/lib.rs]: it's just an extraction from __archive_fetch

File: whisper/src/lib.rs (hunk 2 above)
Comment (andy128k) [reply whisper/src/lib.rs]: You is kinda right. This is how it is implemented in python. My idea was to extract this piece of wisdom to separate function and make real API more straight. 

Now, when this function is on display everyone can see it is ridiculous.

**Pull request merged**